code/game/objects/structures/inflatable.dm

```diff
@@ -0,0 +1,275 @@
+	desc = "A folded membrane which rapidly expands into a large cubical shape on activation."
+	icon_state = "folded_wall"
+	deploy_path = /obj/structure/inflatable/wall
+
+/obj/item/inflatable/door
+	name = "inflatable door"
+	desc = "A folded membrane which rapidly expands into a simple door on activation."
+	icon_state = "folded_door"
+	deploy_path = /obj/structure/inflatable/door
+
+/obj/structure/inflatable
+	name = "inflatable"
+	desc = "An inflated membrane. Do not puncture."
+	density = 1
+	anchored = 1
+	opacity = 0
+	icon = 'icons/obj/inflatable.dmi'
+	icon_state = "wall"
+	penetration_dampening = 3
+
+	var/undeploy_path = null
+	var/health = 50
+
+/obj/structure/inflatable/wall
+	name = "inflatable wall"
+	undeploy_path = /obj/item/inflatable/wall
+
+/obj/structure/inflatable/New(location)
+	..()
+	update_nearby_tiles(need_rebuild=1)
+
+/obj/structure/inflatable/Destroy()
+	update_nearby_tiles()
+	..()
+
+/obj/structure/inflatable/Cross(atom/movable/mover, turf/target, height=0, air_group=0)
+	return 0
+
+/obj/structure/inflatable/bullet_act(var/obj/item/projectile/Proj)
+	health -= Proj.damage
+	..()
+	if(health <= 0)
+		deflate(1)
+
+/obj/structure/inflatable/projectile_check()
+	return PROJREACT_WINDOWS
+
+/obj/structure/inflatable/ex_act(severity)
+	switch(severity)
+		if(1)
+			qdel(src)
+		if(2)
+			deflate(1)
+		if(3)
+			if(prob(50))
+				deflate(1)
+
+/obj/structure/inflatable/attack_hand(mob/user)
+	add_fingerprint(user)
+
+/obj/structure/inflatable/attackby(obj/item/weapon/W, mob/user)
+	if(W.is_sharp())
+		visible_message("<span class='danger'>[user] pierces [src] with [W]!</span>")
+		deflate(1)
+	if(W.damtype == BRUTE || W.damtype == BURN)
+		hit(W.force)
+		..()
+
+/obj/structure/inflatable/attack_alien(mob/user)
+	user.delayNextAttack(10)
+	if(islarva(user))
+		return
+	visible_message("<span class='danger'>\The [user] rips [src] apart!</span>")
+	deflate(1)
+
+/obj/structure/inflatable/attack_animal(mob/user)
+	user.delayNextAttack(10)
+	var/mob/living/simple_animal/M = user
+	if(M.melee_damage_upper <= 0)
+		return
+	user.visible_message("<span class='danger'>\The [user] attacks [src]!</span>")
+	attack_generic(M, M.melee_damage_upper)
+
+/obj/structure/inflatable/attack_slime(mob/user)
+	user.delayNextAttack(10)
+	if(!isslimeadult(user))
+		return
+	user.visible_message("<span class='danger'>\The [user] glomps [src]!</span>")
+	attack_generic(user, rand(10, 15))
+
+/obj/structure/inflatable/proc/hit(var/damage, var/sound_effect = 1)
+	health = max(0, health - damage)
+	if(sound_effect)
+		playsound(loc, 'sound/effects/attackblob.ogg', 75, 1)
+	if(health <= 0)
+		deflate(1)
+
+/obj/structure/inflatable/CtrlClick()
+	hand_deflate()
+
+/obj/structure/inflatable/proc/deflate(var/violent=0)
+	playsound(loc, 'sound/machines/hiss.ogg', 75, 1)
+	if(violent)
+		visible_message("\The [src] rapidly deflates!")
+		var/obj/item/inflatable/torn/R = new /obj/item/inflatable/torn(loc)
+		transfer_fingerprints_to(R)
```

Comment:
GitHub marked my comments as done but you should still move the following down to outside the if/else to avoid copy paste:

``` DM
transfer_fingerprints_to(R)
qdel(src)
```